mobile/src/main/java/rs/readahead/washington/mobile/views/activity/camera/CameraActivity.kt

```diff
@@ -454,11 +461,30 @@ class CameraActivity : MetadataActivity(), IMetadataAttachPresenterContract.IVie
         switchButton.contentDescription = getString(contentDescriptionResId)
     }
 
-    fun onPreviewClicked() {
-        val intent = Intent(this, MainActivity::class.java)
-        intent.putExtra(MainActivity.PHOTO_VIDEO_FILTER, "filter")
-        startActivity(intent)
-        finish()
+    private fun onPreviewClicked() {
+        var intent: Intent? = null
+        lastMediaFile?.mimeType?.let {
+            when {
+                MediaFile.isImageFileType(it) -> {
+                    intent = Intent(this, PhotoViewerActivity::class.java).apply {
```

Comment:
maybe we should create  a separate function createIntentForMediaFile() to encapsulate the logic for creating the intent based on the media file and then called on the fonction onPreviewClicked() like this private fun onPreviewClicked() {
    val intent: Intent? = createIntentForMediaFile()

    intent?.let {
        startActivity(intent)
    }
}